### PR TITLE
Premint - if no mint recipient, revert with proper error

### PR DESCRIPTION
**Pull request by oveddan**

### CODE REVIEW

**Comment by oveddan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #310** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/310" target="_blank">Graphite</a> 
    * **PR #305** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/305" target="_blank">Graphite</a> 
      * **PR #307** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/307" target="_blank">Graphite</a> 
      * **PR #262** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/262" target="_blank">Graphite</a> 
        * **PR #299** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/299" target="_blank">Graphite</a> 
          * **PR #312** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/312" target="_blank">Graphite</a>  👈
            * **PR #289** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/289" target="_blank">Graphite</a> 
          * **PR #303** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/303" target="_blank">Graphite</a> 
            * **PR #309** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/309" target="_blank">Graphite</a> 
        * **PR #294** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/294" target="_blank">Graphite</a> 
          * **PR #295** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/295" target="_blank">Graphite</a> 
        * **PR #273** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/273" target="_blank">Graphite</a> 
          * **PR #288** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/288" target="_blank">Graphite</a> 
            * **PR #284** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/284" target="_blank">Graphite</a> 
          * **PR #280** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/280" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/312?utm_source=stack-comment).

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImplLib.sol [hunk 1]
```diff
@@ -120,7 +120,8 @@ library ZoraCreator1155PremintExecutorImplLib {
         uint256 quantityToMint,
         IZoraCreator1155PremintExecutor.MintArguments memory mintArguments
     ) internal {
-        bytes memory mintSettings = abi.encode(mintArguments.mintRecipient, mintArguments.mintComment);
+        address mintRecipient = mintArguments.mintRecipient == address(0) ? msg.sender : mintArguments.mintRecipient;
```
Comment (iainnash) [inline packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImplLib.sol:123]: In what case would this be true?

**Review by iainnash — COMMENTED** [review]
Just trying to understand the motivation for this pr.

**Comment by oveddan** [comment]
> Just trying to understand the motivation for this pr.

In case it's not set by accident

**Comment by kulkarohan** [comment]
hmm i wonder if its better to leave this explicit and let it revert if a recipient isn't passed

**Review by iainnash — DISMISSED** [review]

**Review by IsabellaSmallcombe — DISMISSED** [review]

**Comment by oveddan** [comment]
### Merge activity

* **Nov 15, 2:48 PM**: @oveddan started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/312).

**Pull request merged**